### PR TITLE
SOC-2304 | Admin Tools: Discussion Header

### PR DESCRIPTION
**Pull request by mleczey**
## Links
* https://wikia-inc.atlassian.net/browse/SOC-1538
* https://wikia-inc.atlassian.net/browse/SOC-2304

## Description

As an admin, I want to change the Discussions Header image for discussions

## Reviewers

@bkoval

### CODE REVIEW

File: front/main/app/components/discussion-community-badge.js [hunk 1]
```diff
@@ -1,41 +1,29 @@
...
 import {track, trackActions} from '../utils/discussion-tracker';
-import EscapePress from '../mixins/escape-press';
 
 export default Ember.Component.extend(
-	EscapePress,
+	DiscussionEditImage,
 	{
 		classNames: ['community-badge', 'draggable-dropzone'],
-		classNameBindings: ['isEditMode', 'isNewBadgePreviewMode', 'isDragActive:drag-activated',
+		classNameBindings: ['isEditMode', 'isImagePreviewMode', 'isDragActive:drag-activated',
```
Comment (bkoval) [inline front/main/app/components/discussion-community-badge.js]: Could "isImagePreviewMode" be moved to the DiscussionEditImage mixin classNameBindings?

File: front/main/app/components/discussion-hero-unit.js [hunk 2]
```diff
@@ -1,161 +1,76 @@
...
-		6840: 'discussion-header-fr-zelda.jpg',
-		1429796: 'discussion-header-pokemon-go.png',
-	},
+		canEdit: Ember.computed.and('editingPossible', 'currentUser.isAuthenticated', 'heroImage.permissions.canEdit'),
+		currentUser: Ember.inject.service(),
+		editingPossible: false,
+		fileInputClassNames: ['upload-image-button', 'background-theme-color'],
+		imageBackground: null,
+		onImageUrlChange: Ember.observer('imageUrl', function () {
+			this.setImageBackground(this.get('imageUrl'));
```
Comment (bkoval) [inline front/main/app/components/discussion-hero-unit.js]: Maybe we could use "imageBackground" as a computed property to get rid of this observer and setImageBackground method?

File: front/main/app/components/discussion-hero-unit.js [hunk 3]
```diff
@@ -1,161 +1,76 @@
...
 
-		if (!this.get('style') && isShown && image) {
-			this.set('style',
-				new Ember.Handlebars.SafeString(`background: #fff url(/front/common/images/${image}) center no-repeat;`));
-			this.set('contentClassNames', 'background-alpha-theme-color');
-		}
-	}),
-});
+		/**
+		 * @private
```
Comment (bkoval) [inline front/main/app/components/discussion-hero-unit.js]: I think this JSDoc lacks return type and a param description.

File: front/main/app/mixins/discussion-edit-image.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+					track(this.get('trackedActions.EditFilePasted'));
+					event.preventDefault();
+					this.send('fileUpload', files);
+				}
+			}
+		},
+
+		// components using this mixin should provide upload method
+		uploadMethod() {
```
Comment (bkoval) [inline front/main/app/mixins/discussion-edit-image.js]: Maybe we should throw here to ensure the method is being overwritten?

File: front/main/app/templates/components/discussion-hero-unit.hbs [hunk 5]
```diff
@@ -1,21 +1,84 @@
...
-			{{discussion-contributors
-				users=contributors.users
-				maxContributorDisplayCount=6
-				totalContributorCount=contributors.count
+<div class="discussion-hero-unit-image {{if isEditMode 'highlight-overlay-content'}}" style="{{imageBackground}}">
+	{{#unless isEditMode}}
+		<div class="discussion-hero-unit-content {{contentClassNames}}">
+			{{#if userName}}
+				{{discussion-post-list-header
+					userName=userName
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-hero-unit.hbs]: Alph order :)

File: front/main/app/templates/components/discussion-hero-unit.hbs [hunk 6]
```diff
@@ -1,21 +1,84 @@
...
+				}}
+			{{else}}
+				<h1>{{headerTitle}}</h1>
+				{{#if numberOfPosts}}
+					<p>{{i18n 'main.number-of-posts' numberOfPosts=numberOfPosts ns='discussion'}}</p>
+				{{/if}}
+				{{#if contributors}}
+					{{discussion-contributors
+						users=contributors.users
+						maxContributorDisplayCount=6
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-hero-unit.hbs]: Also alph

File: front/main/app/templates/components/discussion-hero-unit.hbs [hunk 7]
```diff
@@ -1,21 +1,84 @@
...
+						totalContributorCount=contributors.count
+					}}
+				{{/if}}
+			{{/if}}
+		</div>
+	{{/unless}}
+	{{#if canEdit}}
+		{{#if isEditMode}}
+			{{#if isDragActive}}
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-hero-unit.hbs]: Do you think It'd be possible to combine these three conditions into one computed property producing a boolean variable?

File: front/main/app/mixins/discussion-edit-image.js (hunk 4 above)
Comment (mleczey) [inline front/main/app/mixins/discussion-edit-image.js]: done

**Comment by bkoval** [comment]
LGTM 🚓 👮

**Pull request merged**